### PR TITLE
refactor: Refactor root particle writer and simulation monitoring

### PR DESCRIPTION
**Pull request by andiwand**
Instead of writing two root files with the initial and final state of the simulation particles I think it is beneficial to only store the initial state and some extra information like energy loss, encountered material and number of hits.

This should also make physmon for the simulation more comprehensible and adds important information like hits as a function of eta.

### CODE REVIEW

**Comment by andiwand** [comment]
I wanted to have a hits vs eta plot too but this is currently not possible with the generic plotter and I don't have an easy solution at hand so I will leave this out for now

File: Examples/Io/Root/include/ActsExamples/Io/Root/RootParticleWriter.hpp [hunk 1]
```diff
@@ -37,6 +38,10 @@ class RootParticleWriter final : public WriterT<SimParticleContainer> {
   struct Config {
     /// Input particle collection to write.
     std::string inputParticles;
+    /// Optional. Input final particle collection to write.
```
Comment (benjaminhuth) [inline Examples/Io/Root/include/ActsExamples/Io/Root/RootParticleWriter.hpp:41]:
```suggestion
    /// Optional. If given, the the energy loss and traversed material is computed and written.
```

File: Examples/Io/Root/include/ActsExamples/Io/Root/RootParticleWriter.hpp [hunk 2]
```diff
@@ -37,6 +38,10 @@ class RootParticleWriter final : public WriterT<SimParticleContainer> {
...
+    std::string inputFinalParticles;
+    /// Optional. Input sim hit collection to write.
```
Comment (benjaminhuth) [inline Examples/Io/Root/include/ActsExamples/Io/Root/RootParticleWriter.hpp:43]:
```suggestion
    /// Optional. If given, the number of measurements is computed and written.
```

**Review by benjaminhuth — COMMENTED** [review]
Looks good! I just propose to improve the documentation of the input keys.

**Review by benjaminhuth — APPROVED** [review]
Let's get it in!

**Review by benjaminhuth — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [169fad4a8eb3ceb7f7bf7419d12b57fb2470cc16]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6547828)!

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327654) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327653) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327652) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327651) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327650) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327649) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327648) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327647) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327646) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327645) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327644) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327643) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327642) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327641) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327640) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327639) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327638) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327637) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327636) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34327635) |

**Pull request merged**